### PR TITLE
Vector DB collecion strategy

### PR DESCRIPTION
**Pull request by abhishekgml1991**
## Description

[Provide a brief description of the changes in this PR]

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

## Summary by CodeRabbit

* **New Features**
  * Added configurable vector collection strategies, including per-connector-type collections.
  * Improved multi-collection search with parallel fan-out, result merging, and access-aware filtering.
  * Added organization-scoped duplicate detection and safer vector membership management.
* **Bug Fixes**
  * Administrators or connector creators can delete connectors within their organization.
  * Prevented cross-organization deduplication and unsafe cleanup of active records.
  * Improved collection health checks and rebuild handling.
* **Upgrade Notes**
  * Collection strategy is selected on first startup; changing it later requires an embedding rebuild.

### CODE REVIEW

File: backend/python/app/utils/chat_helpers.py [hunk 1]
```diff
@@ -2940,52 +2962,61 @@ async def create_record_from_vector_metadata(metadata: dict[str, Any], org_id: s
...
+                    # from what we have and say plainly that it is partial.
+                    logger.error(
+                        "Vector scroll for %s in %s stopped early after %d points: %s",
+                        virtual_record_id,
+                        collection_name,
+                        len(points),
+                        e,
+                    )
+                    break
+                points.extend(result.points)
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/app/utils/chat_helpers.py:2992]: can you check if this is fixed

File: backend/python/tests/unit/connectors/api/test_router_part1.py [hunk 2]
```diff
@@ -646,28 +664,41 @@ def test_personal_connector_creator_passes(self):
...
+    def test_personal_connector_admin_passes(self):
+        """Deletion is uniform across scopes: an administrator can remove a
+        personal connector whose creator is gone. Reading or altering it stays
+        blocked by _can_access_connector — a different act."""
+        _validate_connector_deletion_permissions(
+            {"scope": ConnectorScope.PERSONAL.value, "createdBy": "user-a"},
+            "user-b",
+            is_admin=True,
+            logger=MagicMock(),
+        )
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/tests/unit/connectors/api/test_router_part1.py:676]: can you check if this is fixed

File: backend/python/app/services/graph_db/neo4j/neo4j_client.py [hunk 3]
```diff
@@ -161,9 +211,23 @@ async def disconnect(self) -> None:
...
+
+            for driver in drivers:
+                try:
+                    # A driver bound to another (possibly already-stopped) loop
+                    # cannot be closed from here; log and move on rather than
+                    # abandon the remaining ones.
+                    await driver.close()
+                except Exception as e:
+                    self.logger.warning(f"Error closing a Neo4j driver: {str(e)}")
+            if drivers:
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/app/services/graph_db/neo4j/neo4j_client.py]: check now

File: backend/python/app/modules/indexing/run.py [hunk 4]
```diff
@@ -111,12 +141,172 @@ async def rewrite_or_delete_vector_membership(self, virtual_record_id: str) -> s
...
+                        action=scope.action,
+                        collection_names=scope.collection_names,
+                        filter_field=CONNECTOR_IDS_FIELD,
+                        filter_values=(
+                            [ctx.connector_id] if ctx.connector_id else None
+                        ),
+                    )
+                )
+            for name in scope.collection_names:
+                await self.collection_registry.delete_collection(name)
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/app/modules/indexing/run.py:225]: check now

File: backend/python/tests/unit/connectors/api/test_router_part1.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/tests/unit/connectors/api/test_router_part1.py:676]: check now

**Review by abhishek-pipeshub — DISMISSED** [review]

**Review by abhishek-pipeshub — APPROVED** [review]
